src/hooks/root-hooks/use-storage-analytics.ts

```diff
@@ -0,0 +1,54 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { useEffect } from 'react';
+
+import { AnalyticsEventCategory } from 'src/utils/analytics/analytics-event.enum';
+import { useAnalytics } from 'src/utils/analytics/use-analytics.hook';
+import { calculateStringSizeInBytes } from 'src/utils/string.utils';
+
+export const useStorageAnalytics = () => {
+  const { trackEvent } = useAnalytics();
+
+  useEffect(
+    () =>
+      void getAsyncStorageUsageDetails().then(details =>
+        trackEvent('STORAGES_STATE', AnalyticsEventCategory.General, { asyncStorage: details })
+      ),
+    [trackEvent]
+  );
+};
+
+const getAsyncStorageUsageDetails = async () => {
+  const keys = await AsyncStorage.getAllKeys();
+
+  const records = await Promise.all(
+    keys.map(key =>
+      AsyncStorage.getItem(key).then(
+        val => [key, val] as const,
+        error => {
+          console.error(error);
```

Comment:
Could you, please, clarify one thing? AFAIU the issue we have been trying to track is related to AsyncStorage not being able to `setItem`,  but not `getItem`. So if the key is "oversized" as you call it, we won't be able to set it at all and most likely won't catch the error here 🤔

Reply:
Ok! Thank you for clarification